src/Core/Content/Product/Cms/ProductSlider/ProductStreamProcessor.php

```diff
@@ -0,0 +1,191 @@
+
+        return $collection;
+    }
+
+    public function enrich(CmsSlotEntity $slot, ElementDataCollection $result, ResolverContext $resolverContext): void
+    {
+        $entitySearchResult = $result->get(self::PRODUCT_SLIDER_ENTITY_FALLBACK . '_' . $slot->getUniqueIdentifier());
+        if (!$entitySearchResult) {
+            return;
+        }
+
+        $streamResult = $entitySearchResult->getEntities();
+        if (!$streamResult instanceof ProductCollection) {
+            return;
+        }
+
+        $slider = new ProductSliderStruct();
+        $slot->setData($slider);
+
+        $slider->setProducts(
+            $this->handleProductStream(
+                $streamResult,
+                $resolverContext->getSalesChannelContext(),
+                $entitySearchResult->getCriteria()
+            )
+        );
+
+        $config = $slot->getFieldConfig();
+
+        $productConfig = $config->get('products');
+        \assert($productConfig instanceof FieldConfig);
+
+        $slider->setStreamId($productConfig->getStringValue());
+    }
+
+    private function collectByProductStream(
+        ResolverContext $resolverContext,
+        FieldConfig $config,
+        FieldConfigCollection $elementConfig
+    ): Criteria {
+        $filters = $this->productStreamBuilder->buildFilters(
+            $config->getStringValue(),
+            $resolverContext->getSalesChannelContext()->getContext()
+        );
+
+        $limit = $elementConfig->get('productStreamLimit')?->getIntValue() ?? self::FALLBACK_LIMIT;
+
+        $criteria = new Criteria();
+        $criteria->addFilter(...$filters);
+        $criteria->setLimit($limit);
+
+        $this->addGrouping($criteria);
+        $sorting = $elementConfig->get('productStreamSorting')?->getStringValue() ?? 'name:' . FieldSorting::ASCENDING;
+
+        if ($sorting === 'random') {
+            $this->addRandomSort($criteria);
+        } else {
+            $sorting = explode(':', $sorting);
+            $field = $sorting[0];
+            $direction = $sorting[1];
+
+            $criteria->addSorting(new FieldSorting($field, $direction));
+        }
+
+        return $criteria;
+    }
+
+    private function handleProductStream(
+        ProductCollection $streamResult,
+        SalesChannelContext $context,
+        Criteria $originCriteria
+    ): ProductCollection {
+        $finalProductIds = $this->collectFinalProductIds($streamResult);
+        if (\count($finalProductIds) === 0) {
+            return new ProductCollection();
+        }
+
+        $criteria = $originCriteria->cloneForRead($finalProductIds);
+
+        $products = $this->productRepository->search($criteria, $context)->getEntities();
+        \assert($products instanceof ProductCollection);
+        $products->sortByIdArray($finalProductIds);
+
+        return $products;
+    }
+
+    /**
+     * @return string[] List of product ids
```

Comment:
```suggestion
     * @return list<string> List of product ids
```

if even the comment says it is a list :wink: